src/js/hooks/cycleCount/useResolveStep.js

```diff
@@ -43,17 +43,41 @@ const useResolveStep = () => {
     currentLocation: state.session.currentLocation,
   }));
 
+  const mergeCycleCountItems = (items) => {
+    const duplicatedItems = _.groupBy(items,
+      (item) => `${item.binLocation?.id}-${item?.inventoryItem?.lotNumber}`);
+    return Object.values(duplicatedItems).map((itemsToMerge) => {
+      const maxCountIndex = _.maxBy(itemsToMerge, 'countIndex').countIndex;
+      const itemFromCount = _.find(itemsToMerge, (item) => item.countIndex === maxCountIndex - 1);
+      const itemFromResolve = _.find(itemsToMerge, (item) => item.countIndex === maxCountIndex);
+      return itemFromCount ? {
+        ...itemFromResolve,
+        ...itemFromCount,
```

Comment:
If these `mergeCycleCountItems` will be used in the `save` action, then we might always get the `quantityCounted` from the count item instead of the recount item (unless later `quantityRecounted` is used for building request payload, then it is ok and you can disregard this comment).

Reply:
yeah, there will be a need for building a new payload.